src/main/java/org/springframework/samples/petclinic/owner/PetController.java

```diff
@@ -87,6 +88,10 @@ public String processCreationForm(Owner owner, @Valid Pet pet, BindingResult res
 		if (StringUtils.hasLength(pet.getName()) && pet.isNew() && owner.getPet(pet.getName(), true) != null) {
 			result.rejectValue("name", "duplicate", "already exists");
 		}
+		LocalDate currentDate = LocalDate.now();
+		if (pet.getBirthDate() != null && pet.getBirthDate().isAfter(currentDate)) {
+			result.rejectValue("birthDate", "invalid", "Invalid birth date. Please select a valid date.");
```

Comment:
The error code should probably be more unique than simply "invalid"? Maybe "invalid.date" (and then add a message to `messages.properties`).

Reply:
okay now its done you can check it in the new commit